### PR TITLE
remove nlp/modules

### PR DESCRIPTION
**Pull request by dimapihtar**
> [!IMPORTANT]  
> The `Update branch` button must only be pressed in very rare occassions.
> An outdated branch is never blocking the merge of a PR.
> Please reach out to the automation team before pressing that button.

# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog

- Add specific line by line info of high level changes in this PR.

# Usage

- You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"

**Pre checks**:

- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:

- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed.
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information

- Related to # (issue)

### CODE REVIEW

File: nemo/collections/speechlm/utils/text_generation/audio_text_generation_utils.py [hunk 1]
```diff
@@ -15,18 +15,18 @@
...
 import pickle
+import sys
 from collections.abc import Iterable
 from typing import List, Optional, Tuple, Union
 import numpy as np
 import torch
 import torch.nn.functional as F
 
-import nemo.collections.nlp.modules.common.text_generation_utils as text_generation_utils
+import nemo.collections.multimodal.speech_llm.modules.common.text_generation_utils as text_generation_utils
```
Comment (nithinraok) [inline nemo/collections/speechlm/utils/text_generation/audio_text_generation_utils.py:24]: isn;t `multimodal` also getting removed?

File: nemo/collections/speechlm/utils/text_generation/audio_text_generation_utils.py [hunk 2]
```diff
@@ -47,6 +47,11 @@
         _reconfigure_microbatch_calculator as reconfigure_num_microbatches_calculator,
     )
 
+if sys.version_info >= (3, 8):
+    from typing import TypedDict
+else:
+    from typing_extensions import TypedDict
+
```
Comment (nithinraok) [inline nemo/collections/speechlm/utils/text_generation/audio_text_generation_utils.py]: Do we need this? As per https://github.com/NVIDIA-NeMo/NeMo?tab=readme-ov-file#requirements min version for python is 3.10

File: nemo/collections/speechlm/utils/text_generation/audio_text_generation_utils.py (hunk 1 above)
Comment (dimapihtar) [reply nemo/collections/speechlm/utils/text_generation/audio_text_generation_utils.py:24]: hmm not aware about that. Let's keep it here so far I think. We can move it in a separate PR if we're going to remove the multimodal modules.
